C/include/c4ConnectedClient.h

```diff
@@ -37,15 +68,47 @@ C4ConnectedClient* c4client_new(const C4ConnectedClientParameters* params,
  @param asFleece  If true, the response's `body` field is Fleece; if false, it's JSON.
  @param callback Callback for getting document.
  @param context Client value passed to getDocument callback
- @param error  On failure, the error info will be stored here. */
+ @param outError  On failure, the error info will be stored here. */
 void c4client_getDoc(C4ConnectedClient*,
                      C4Slice docID,
                      C4Slice collectionID,
                      C4Slice unlessRevID,
                      bool asFleece,
                      C4ConnectedClientGetDocumentCallback callback,
                      void * C4NULLABLE context,
-                     C4Error* error) C4API;
+                     C4Error* outError) C4API;
+
+/** Callback for updating the document result.
+ 
+ @param client   The client that initiated the callback.
+ @param err Error will be written here if the get-document fails.
+ @param context  user-defined parameter given when registering the callback. */
+typedef void (*C4ConnectedClientUpdateDocumentCallback)(C4ConnectedClient* client,
+                                                        C4HeapSlice revID,
+                                                        C4Error* C4NULLABLE err,
+                                                        void * C4NULLABLE context);
+
+typedef C4ConnectedClientUpdateDocumentCallback C4ConnectedClientUpdateDocumentCallback;
+
+/** Pushes a new document revision to the server.
+ @param docID  The document ID.
+ @param collectionID  The name of the document's collection, or `nullslice` for default.
+ @param revID  The ID of the parent revision on the server,
+                      or `nullslice` if this is a new document.
+ @param revisionFlags  Flags of this revision.
+ @param fleeceData  The document body encoded as Fleece (without shared keys!)
+ @param callback Callback once the document is updated.
+ @param context Client value passed to updateDocument callback
+ @param outError  On failure, the error info will be stored here. */
+void c4client_updateDoc(C4ConnectedClient* client,
```

Comment:
Functions that can fail shouldn't return void. It should return bool to indicate whether there's an error.